### PR TITLE
Expand documentation for patching generics

### PR DESCRIPTION
**Pull request by abrenneke**
Expand the documentation for patching generic types based on information learned from Mono blog posts, documentation, and experimentation.

### CODE REVIEW

**Comment by abrenneke** [comment]
Whoops my editor automatically removed empty whitespace. Can revert that if you want.

**Comment by pardeike** [comment]
Unfortunately, you edited the auto-generated file for this [source](https://github.com/pardeike/Harmony/blob/master/Harmony/Documentation/articles/patching-edgecases.md).

**Comment by abrenneke** [comment]
Oops thanks

**Comment by abrenneke** [comment]
@pardeike markdown file edited

File: docs/articles/patching-edgecases.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<!DOCTYPE html>
```
Comment (pardeike) [inline docs/articles/patching-edgecases.html:1]: This file will be overwritten by the CI bot that auto generates the documentation from the above markdown file.

**Pull request merged**